app/javascript/components/server-components/Discover/index.tsx

```diff
@@ -253,12 +298,19 @@ const Discover = (props: Props) => {
       taxonomyPath={taxonomyPath}
       taxonomiesForNav={props.taxonomies_for_nav}
       showTaxonomy
-      onTaxonomyChange={(newTaxonomyPath) =>
+      onTaxonomyChange={(newTaxonomyPath) => {
+        // Read from URL to avoid stale state when user clicks Clear then immediately navigates
```

Comment:
Had to do this to avoid a state management issue : 

1. Click on Get Black Friday Deals (?offer_code=BLACKFRIDAY2025)
2. Change taxonomy (ex: Click on 3D)
3. Remove the offer code (Click on "Clear")
4. Change taxonomy again (ex: Click on Audio), still appends ?offer_code=BLACKFRIDAY2025 although it shouldn't